File: LeetCode/3274-check-if-two-chessboard-squares-have-the-same-color/3274-check-if-two-chessboard-squares-have-the-same-color.cpp
class Solution {
public:
    char solve(string s) {
        if ((s[1] - '0') % 2 == 1) {
            int t = s[0] - 'a';
            if (t % 2 == 1) {
                return 'w';
            } else {
                return 'b';
            }
        } else {
            int t = s[0] - 'a';
            if (t % 2 == 1) {
                return 'b';
            } else {
                return 'w';
            }
        }
    }
    bool checkTwoChessboards(string coordinate1, string coordinate2) {
        char c1, c2;
        c1 = solve(coordinate1);
        c2 = solve(coordinate2);
        return c1 == c2;
    }
};

File: LeetCode/3274-check-if-two-chessboard-squares-have-the-same-color/3274-check-if-two-chessboard-squares-have-the-same-color.py
class Solution:
    def solve(self, s: str) -> str:
        row = int(s[1])
        col = ord(s[0]) - ord('a')
        if row % 2 == 1:
            if col % 2 == 1:
                return 'w'
            else:
                return 'b'
        else:
            if col % 2 == 1:
                return 'b'
            else:
                return 'w'
    
    def checkTwoChessboards(self, coordinate1: str, coordinate2: str) -> bool:
        color1 = self.solve(coordinate1)
        color2 = self.solve(coordinate2)
        return color1 == color2